Header header
uint32 height
uint32 width
CircleInfo[] circles